### PR TITLE
docs: initialize a guide to use the V2 contracts in localnet

### PR DESCRIPTION
**Pull request by lumtis**
Closes: https://github.com/zeta-chain/protocol-contracts/issues/234

This PR adds a document introducing the localnet environment for V2 contracts.

Over the time, the documentation and local environment tooling will be improved and potentially moved to a new repo outlined here: https://github.com/zeta-chain/protocol-contracts/issues/224

This document allows to have an initial entrypoint for developers who want to experiment with the new architecture

## Summary by CodeRabbit

- **Documentation**
  - Added documentation for V2 Contracts - Local Development Environment, explaining the architecture for building Universal Apps with GatewayZEVM and GatewayEVM. Includes functions for depositing, withdrawing, and cross-chain smart contract calls.
  - Provides instructions for deploying a local network using Anvil and includes examples of contracts utilizing the V2 interface.

### CODE REVIEW

File: v2_localnet.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+function withdrawAndCall(bytes memory receiver, uint256 amount, address zrc20, bytes calldata message)
+```
+
+* Simple call to a contract on a connected chain
+
+```
+function call(bytes memory receiver, bytes calldata message) external
+```
+
+## Experimenting with the New Architecture
```
[the remark this answers is not in the document]
Comment (lumtis) [reply v2_localnet.md:81]: Using title case for the section title

File: v2_localnet.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+Running the command will deploy the two gateway contracts:
+
+```
+[WORKER] GatewayEVM: 0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512
+[WORKER] GatewayZEVM: 0x0165878A594ca255338adfa4d48449f69242Eb8F
+```
+
+The developers can develop application using these addresses, the messages will automatically be relayed by the worker.
+
+The local environment uses Anvil. Therefore, the default Anvil accounts can be used to interact with the network.
```
[the remark this answers is not in the document]
Comment (lumtis) [reply v2_localnet.md:112]: Nope, will revert this

File: v2_localnet.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+```
+Account #0: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 (10000 ETH)
+Private Key: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
+
+Account #1: 0x70997970C51812dc3A010C7d01b50e0d17dc79C8 (10000 ETH)
+Private Key: 0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d
+
+Account #2: 0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC (10000 ETH)
+Private Key: 0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a
+```
```
[the remark this answers is not in the document]
Comment (lumtis) [reply v2_localnet.md:123]: It's test account this is fine

File: v2_localnet.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+* Call: Smart contract call involved in cross-chain transactions.
+
+
+In consequence, the interface is as follow:
+* `GatewayEVM`: `deposit`, `depositAndCall`, and `call`
+* `GatewayZEVM`: `withdraw`, `withdrawAndCall`, and `call`
+
+The motivation behind this interface is intuitiveness and simplicity. We support different asset types by using function overloading.
+
+### `GatewayEVM`
```
Comment (fbac) [inline v2_localnet.md:29]: nit: when documenting contracts we can consider leveraging slither printers.

i.e:

The command `slither contracts/prototypes/zevm/GatewayZEVM.sol --print contract-summary` will output the following, which we can reuse:

```
+ Contract GatewayZEVM (Upgradeable) (Most derived contract)
  - From UUPSUpgradeable
    - __UUPSUpgradeable_init() (internal)
    - __UUPSUpgradeable_init_unchained() (internal)
    - proxiableUUID() (external)
    - upgradeTo(address) (public)
    - upgradeToAndCall(address,bytes) (public)
  - From ERC1967UpgradeUpgradeable
    - __ERC1967Upgrade_init() (internal)
    - __ERC1967Upgrade_init_unchained() (internal)
    - _changeAdmin(address) (internal)
    - _getAdmin() (internal)
    - _getBeacon() (internal)
    - _getImplementation() (internal)
    - _setAdmin(address) (private)
    - _setBeacon(address) (private)
    - _setImplementation(address) (private)
    - _upgradeBeaconToAndCall(address,bytes,bool) (internal)
    - _upgradeTo(address) (internal)
    - _upgradeToAndCall(address,bytes,bool) (internal)
    - _upgradeToAndCallUUPS(address,bytes,bool) (internal)
  - From Initializable
    - _disableInitializers() (internal)
    - _getInitializedVersion() (internal)
    - _isInitializing() (internal)
  - From OwnableUpgradeable
    - __Ownable_init() (internal)
    - __Ownable_init_unchained() (internal)
    - _checkOwner() (internal)
    - _transferOwnership(address) (internal)
    - owner() (public)
    - renounceOwnership() (public)
    - transferOwnership(address) (public)
  - From ContextUpgradeable
    - __Context_init() (internal)
    - __Context_init_unchained() (internal)
    - _contextSuffixLength() (internal)
    - _msgData() (internal)
    - _msgSender() (internal)
  - From GatewayZEVM
    - _authorizeUpgrade(address) (internal)
    - _withdraw(uint256,address) (internal)
    - call(bytes,bytes) (external)
    - constructor() (public)
    - deposit(address,uint256,address) (external)
    - depositAndCall(zContext,address,uint256,address,bytes) (external)
    - execute(zContext,address,uint256,address,bytes) (external)
    - initialize() (public)
    - withdraw(bytes,uint256,address) (external)
    - withdrawAndCall(bytes,uint256,address,bytes) (external)
```

The full list of slither printers:

```
cfg, ck, constructor-calls, contract-summary, data-dependency, declaration, dominator, echidna, function-id, function-summary, halstead, loc, martin, modifiers, call-graph, evm, human-summary, inheritance, inheritance-graph, slithir, slithir-ssa, not-pausable, vars-and-auth, require, variable-order
```

**Review by fbac — APPROVED** [review]

File: v2_localnet.md (hunk 4 above)
Comment (lumtis) [reply v2_localnet.md:29]: Sounds good, just wanted to display the dev interface here

**Pull request merged**